### PR TITLE
Calculates rebar length and divide them in positions

### PR DESCRIPTION
**Pull request by StollToaldo**

### CODE REVIEW

File: structuraldesign/detailing/beam_detailing.py [hunk 1]
```diff
@@ -146,19 +164,20 @@ def _zone_rebars(self, zone, nrebars, fi, al, lb):
...
         #offsets the diagram
-        for i in xrange(0, nrebars):
-            m = mlist[i]
+        for i in xrange(nrebars + 1):
+            m = i * dm
+            m_list.append(m)
 
             #calculates the rebar points of interception (rebar_poi) with the diagram
-            rebar_poi = self.diagram.get_x(m, zone.x0, zone.x1)
+            rebar_poi = self.get_x(round(m,3), zone.x0 - 0.5, zone.x1 + 0.5) # TODO: precision issues
```
Comment (StollToaldo) [inline structuraldesign/detailing/beam_detailing.py]: Take a look at this shit.

File: structuraldesign/detailing/beam_detailing.py [hunk 2]
```diff
@@ -146,19 +164,20 @@ def _zone_rebars(self, zone, nrebars, fi, al, lb):
...
 
+        m_list = []
         all_rebar_poi = []
         all_l_offset = []
         all_r_offset = []
 
         #offsets the diagram
-        for i in xrange(0, nrebars):
-            m = mlist[i]
+        for i in xrange(nrebars + 1):
```
Comment (rodrigorahal) [inline structuraldesign/detailing/beam_detailing.py]: `m_list = [i * dm for i in xrage(nrebars + 1)]`

File: structuraldesign/detailing/beam_detailing.py (hunk 1 above)
Comment (rodrigorahal) [inline structuraldesign/detailing/beam_detailing.py]: As discussed, make a `isclose` function in the utils and use it inside the `get_x` method 👍

File: structuraldesign/detailing/beam_detailing.py [hunk 3]
```diff
@@ -146,19 +164,20 @@ def _zone_rebars(self, zone, nrebars, fi, al, lb):
...
+            m = i * dm
+            m_list.append(m)
 
             #calculates the rebar points of interception (rebar_poi) with the diagram
-            rebar_poi = self.diagram.get_x(m, zone.x0, zone.x1)
+            rebar_poi = self.get_x(round(m,3), zone.x0 - 0.5, zone.x1 + 0.5) # TODO: precision issues
+            print "m = ", m, "rebar_poi = ", rebar_poi
```
Comment (rodrigorahal) [inline structuraldesign/detailing/beam_detailing.py]: Debugging only?
We should learn to use `logging`.
I've tried but didn't quite work.

**Comment by rodrigorahal** [comment]
Great, great job!! Just add the `isclose` and some list comprehension and we're good to go.

**Pull request merged**